src/validation.cpp

```diff
@@ -1354,6 +1335,32 @@ MempoolAcceptResult MemPoolAccept::AcceptSingleTransaction(const CTransactionRef
         return MempoolAcceptResult::Failure(ws.m_state);
     }
 
+    // Now that we've verified the cluster limit is respected, we can perform
+    // calculations involving the full ancestors of the tx.
```

Comment:
It seems a bit unfortunate that, prior to 675fa79bb280129fb87e5d1884e670a35a03f5cd, there isn't a cap on ancestor count when we call `CalculateMemPoolAncestors`. Could the rework be done earlier?

Reply:
I took a stab at it and it's a bit tedious (I managed to move it up earlier in the commit history, but stopped when I got to a conflict with a commit that also touches truc_policy in different ways).  

My overall take is it's not such a big deal, as the concern is not around correctness of the code but a potential CPU DoS if exposed over the network, which doesn't strike me as all that concerning for an intermediate commit.